addOns/paramdigger/src/test/java/org/zaproxy/addon/paramdigger/CacheControllerUnitTest.java

```diff
@@ -0,0 +1,358 @@
+                                params.put(entry.getKey(), entry.getValue().get(0));
+                                try {
+                                    Thread.sleep(70);
+                                } catch (InterruptedException e) {
+                                    e.printStackTrace();
+                                }
+                            }
+                        }
+                        Response resp =
+                                newFixedLengthResponse(
+                                        getHtml(
+                                                "AttributeName.html",
+                                                new String[][] {{"q", ""}, {"p", ""}}));
+                        return resp;
+                    }
+                });
+
+        String url = getHttpMessage(path).getRequestHeader().getURI().toString();
+        config.setUrl(url);
+
+        // When
+        cacheController = new CacheController(this.httpSender, config);
+
+        // Then
+        assertThat(cacheController.isCached(Method.GET), equalTo(true));
+        assertThat(cacheController.getCache().hasTimeIndicator(), equalTo(true));
+        assertThat(cacheController.getCache().isCacheBusterFound(), equalTo(true));
+        assertThat(cacheController.getCache().isCacheBusterIsParameter(), equalTo(true));
+    }
+
+    @Test
+    void shouldFindCacheWithHeaderCacheBusterWhenNoIndicatorPresent()
+            throws HttpMalformedHeaderException {
+        String path = "/testnoindicheader";
+        Map<String, String> headers = new HashMap<>();
+        this.nano.addHandler(
+                new NanoServerHandler(path) {
+                    @Override
+                    protected Response serve(IHTTPSession session) {
+                        Map<String, String> hs = session.getHeaders();
+                        for (Map.Entry<String, String> entry : hs.entrySet()) {
+                            if (!headers.containsKey(entry.getKey())) {
+                                headers.put(entry.getKey(), entry.getValue());
+                                try {
+                                    Thread.sleep(60);
+                                } catch (InterruptedException e) {
+                                    e.printStackTrace();
+                                }
+                            } else if (headers.get(entry.getKey()) != null
+                                    && entry.getValue() != null
+                                    && !headers.get(entry.getKey()).equals(entry.getValue())) {
+                                headers.put(entry.getKey(), entry.getValue());
+                                try {
+                                    Thread.sleep(70);
+                                } catch (InterruptedException e) {
+                                    e.printStackTrace();
+                                }
+                            }
```

Comment:
The other thing I've discovered while testing this method repeatedly/serially is that there a plenty of requests made that don't this either of these conditions. You know your code better than I but if these are the only two conditions you expect to be hit then there's something up. (To see you can just add an `else` block that does a `System.out.println("Fell thru");` or something like that.